app/src/main/java/com/pose/move/feature/auth/register/RegisterViewModel.kt

```diff
@@ -25,25 +25,31 @@ class RegisterViewModel @Inject constructor(
         _uiState.value = _uiState.value.copy(isLoading = true)
 
         viewModelScope.launch {
-            val result = userService.registerUser(
-                RegisterRequest(
-                    _uiState.value.email,
-                    _uiState.value.userName,
-                    _uiState.value.password
-                )
+            val result = userRemoteDataSource.registerUser(
```

Comment:
I would suggest moving all the logic for registering an user into an `*Usecase` class.
That should also be the place where you store your `authToken` and do other logic related to register/login.

ViewModels should be the dumbest components in a project. Their only purpose should be to forward events and responses.

Reply:
I'll solve the useCases issue in the next PR